src/sandstorm/sandstorm-http-bridge.c++

```diff
@@ -1195,7 +1201,59 @@ private:
     auto req = apiCap.restoreRequest();
     req.setToken(percentDecode(buf));
 
-    return req.send().getCap().castAs<Identity>();
+    // Don't directly call `getCap()` on the `RemotePromise` because we want to make sure
+    // that the returned promise fails if the request fails.
+    return req.send().then([](auto response) {
+      return response.getCap().template castAs<Identity>();
+    });
+  }
+
+  void saveIdentityInternal(kj::StringPtr textId, Identity::Client identity) {
+    auto req = apiCap.saveRequest();
+    req.setCap(identity);
+    req.initLabel().setDefaultText("user identity");
+    tasks.add(req.send().then([this,textId](auto result) -> void {
+      // Sandstorm tokens are primarily text but use percent-encoding to be safe.
+      auto tokenText = percentEncode(result.getToken());
+
+      // Clean up any existing symlink.
+      dropIdentity(textId);
```

Comment:
> However, presumably if we're saving over an existing token, it's because the existing token was already observed to be dead?

Yeah, but there's also a nonzero chance that the failure was somehow transient, so we'd like to delay dropping the old sturdyref as long as we can.

We want to:
  1. Avoid leaking sturdyrefs.
  2. Not delete anything in the identities/ directory until we have something to replace it with. Ideally, the replacement would be atomic.
  3. Keep it simple.

> Maybe dropIdentity() should actually be called at the point where this observation was made, rather than here? And if, in the meantime, some other concurrent task has saved a copy of the identity, then we should not overwrite it?

This seems strictly worse for goals (2) and (3).